File: Binary Tree Java/Binary Tree Inorder Traversal/IterativeSolution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res=new ArrayList<>();
        if(root==null)    
            return res;
        
        Stack<TreeNode> S= new Stack<>();
        
        while(root!=null || !S.isEmpty()) {
            if(root!=null) {
                S.add(root);
                root=root.left;
            }
            else {
                root=S.pop();
                res.add(root.val);
                root=root.right;
            }
        }
        
        return res;
    }
}

File: Binary Tree Java/Binary Tree Inorder Traversal/IterativeSolution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun inorderTraversal(root: TreeNode?): List<Int> {
        val res = ArrayList<Int>()
        
        root?: return res
        
        var temp=root
        val st = Stack<TreeNode>()
        
        while(!st.isEmpty() || temp!=null)
            if(temp!=null) {
                st.add(temp)
                temp= temp.left
            }
            else {
                val poped = st.pop()
                res.add(poped.`val`)
                temp=poped.right
            }
            
        return res
    }
}
